### ISSUE TITLE
Tree schema doesn't mark name as mandatory when in remote view container

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#116122]
@jrieken, I am seeing the following error in the OSS console and I believe that it is related to this fix:
_vscode.vscode-api-tests]: property `name` is mandatory and must be of type `string`_

_Originally posted by @lszomoru in https://github.com/microsoft/vscode/issues/115746#issuecomment-775211797_

**Comment by jrieken** [comment]
* have a tree contribution as shown below
* remove the name
* 🐛 no error reported, tho at runtime an error shows

```
"views": {
      "remote": [
        {
          "id": "test.treeId",
          "name": "test-tree",
          "when": "never"
        }
      ]
    }
```


<img width="322" alt="Screenshot 2021-02-08 at 16 51 28" src="https://user-images.githubusercontent.com/1794099/107244441-4eeb1080-6a2e-11eb-8d78-7a54af0832c9.png">

**Comment by alexr00** [comment]
Fixed by 82c57ad1fe4161bb129fb9cb0689e8a7ddb11ff9
